agenta-web/cypress/e2e/ai-critic-evaluation.cy.ts

```diff
@@ -0,0 +1,103 @@
+describe("AI Critics Evaluation workflow", () => {
+    let app_id
+    before(() => {
+        cy.createVariantsAndTestsets()
+        cy.get("@app_id").then((appId) => {
+            app_id = appId
+        })
+    })
+
+    context("When you select evaluation without an API key", () => {
+        beforeEach(() => {
+            cy.visit(`/apps/${app_id}/evaluations`)
+            cy.clearLocalStorage("openAiToken")
+
+            cy.get('[data-cy="evaluation-error-modal"]').should("not.exist")
+            cy.get('[data-cy="ai-critic-button"]').click()
+
+            cy.get('[data-cy="variants-dropdown-0"]').trigger("mouseover")
+            cy.get('[data-cy="variant-0"]').click()
+            cy.get('[data-cy="variants-dropdown-0"]').trigger("mouseout")
+
+            cy.get('[data-cy="selected-testset"]').trigger("mouseover")
+            cy.get('[data-cy="testset-0"]').click()
+            cy.get('[data-cy="selected-testset"]').trigger("mouseout")
+
+            cy.clickLinkAndWait('[data-cy="start-new-evaluation-button"]')
+            cy.get('[data-cy="evaluation-error-modal"]').should("exist")
+        })
+
+        it("Should display when starting evaluation", () => {
+            cy.get('[data-cy="evaluation-error-modal-ok-button"]').click()
+        })
+
+        it("Should navigate to Settings when clicking on the modal", () => {
+            cy.get('[data-cy="evaluation-error-modal-nav-button"]').click()
+            cy.url().should("include", "/settings")
+        })
+    })
+
+    context("When you select evaluation with an API key", () => {
+        beforeEach(() => {
+            cy.visit("/settings")
```

Comment:
let's add "adding the open ai key" in a separate callable test as we will do it in flows